src/Symfony/Bridge/Monolog/Monolog.php

```diff
@@ -0,0 +1,500 @@
+
+        foreach ($this->monolog->getHandlers() as $handler) {
+            if ($handler instanceof DebugLoggerInterface) {
+                return $handler;
+            }
+        }
+
+        return null;
+    }
+
+    public function getName(): string
+    {
+        return $this->monolog->getName();
+    }
+
+    /**
+     * Return a new cloned instance with the name changed.
+     */
+    public function withName(string $name): static
+    {
+        $new = clone $this;
+        $new->monolog = $this->monolog->withName($name);
+
+        return $new;
+    }
+
+    /**
+     * Pushes a handler on to the stack.
+     */
+    public function pushHandler(HandlerInterface $handler): self
+    {
+        $this->monolog->pushHandler($handler);
+
+        return $this;
+    }
+
+    /**
+     * Pops a handler from the stack.
+     *
+     * @throws \LogicException If empty handler stack
+     */
+    public function popHandler(): HandlerInterface
+    {
+        return $this->monolog->popHandler();
+    }
+
+    /**
+     * Set handlers, replacing all existing ones.
+     *
+     * If a map is passed, keys will be ignored.
+     *
+     * @param list<HandlerInterface> $handlers
+     */
+    public function setHandlers(array $handlers): static
+    {
+        $this->monolog->setHandlers($handlers);
+
+        return $this;
+    }
+
+    /**
+     * @return list<HandlerInterface>
+     */
+    public function getHandlers(): array
+    {
+        return $this->monolog->getHandlers();
+    }
+
+    /**
+     * Adds a processor on to the stack.
+     *
+     * @phpstan-param ProcessorInterface|(callable(LogRecord): LogRecord) $callback
+     */
+    public function pushProcessor(ProcessorInterface|callable $callback): static
+    {
+        $this->monolog->pushProcessor($callback);
+
+        return $this;
+    }
+
+    /**
+     * Removes the processor on top of the stack and returns it.
+     *
+     * @phpstan-return ProcessorInterface|(callable(LogRecord): LogRecord)
+     *
+     * @throws \LogicException If empty processor stack
+     */
+    public function popProcessor(): callable
+    {
+        return $this->monolog->popProcessor();
+    }
+
+    /**
+     * @return callable[]
+     *
+     * @phpstan-return array<ProcessorInterface|(callable(LogRecord): LogRecord)>
+     */
+    public function getProcessors(): array
+    {
+        return $this->monolog->getProcessors();
+    }
+
+    /**
+     * Control the use of microsecond resolution timestamps in the 'datetime'
+     * member of new records.
+     *
+     * As of PHP7.1 microseconds are always included by the engine, so
+     * there is no performance penalty and Monolog 2 enabled microseconds
+     * by default. This function lets you disable them though in case you want
+     * to suppress microseconds from the output.
+     *
+     * @param bool $micro True to use microtime() to create timestamps
+     */
+    public function useMicrosecondTimestamps(bool $micro): BaseLogger
```

Comment:
let's return $this; instead everywhere we return BaseLogger currently, there's no need to keep returning BaseLogger if we have a new class.